lib/home_widget.dart

```diff
@@ -97,4 +100,46 @@ class HomeWidget {
     ];
     return _channel.invokeMethod('registerBackgroundCallback', args);
   }
+
+  /// Generate a screenshot based on the build context of a widget.
+  /// This method renders the widget to an image (png) file with the provided filename.
+  /// The png file is saved to the App Group container and the full path is returned as a string.
+  /// The filename is optionally saved to UserDefaults using the provided key.
+  static Future<String?> renderFlutterWidget(
+    String appGroupId,
+    BuildContext context,
+    String filename,
+    String? key,
+  ) async {
+    // Get the render object for the widget
+    final RenderRepaintBoundary boundary =
+        context.findRenderObject() as RenderRepaintBoundary;
+
+    // Create a screenshot of the widget
+    final image = await boundary.toImage(
+        pixelRatio: MediaQuery.of(context).devicePixelRatio);
+    final byteData = await image.toByteData(format: ImageByteFormat.png);
+
+    // Save the screenshot to a file in the app group container
+    final PathProviderFoundation provider = PathProviderFoundation();
+    try {
+      final String? directory = await provider.getContainerPath(
+        appGroupIdentifier: appGroupId,
+      );
+      final String path = '$directory/$filename.png';
+      final File file = File(path);
+      await file.writeAsBytes(byteData!.buffer.asUint8List());
+
+      // Save the filename to UserDefaults if a key was provided
+      if (key != null) {
+        _channel.invokeMethod<bool>('saveWidgetData', {
+          'id': key,
+          'data': '$filename.png',
```

Comment:
I guess this should have the full path in the data, in order to be easier to fully read without having to?

That way it would be less work to actually use it in the Widget on the native side.

Looking at:
https://github.com/leighajarett/homescreen_widgets_example/blob/86ed20a12fb8adfb6569772462bb3df4061f53ee/ios/NewsWidgets/NewsWidgets.swift#L63

I think it would be helpful if this could be slimmed down or possible a method would be provided on the iOS Side of the plugin to help this.